### PR TITLE
LogsDB `host` and `timestamp` mappings tests

### PR DESCRIPTION
**Pull request by salvatore-campagna**
Here we are testing mappings of `host` and `timestamp` fields as they are
used as default fields to sort on when using LogsDB. LogsDB uses a
`host.name` field mapped as a `keyword` and a `@timestamp` field, required
by data streams. Some mappings throw errors as a result of incompatibilities
when trying to merge object fields. Such errors are expected.

See also #113600.

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
Adding such tests is done to capture issues observed by Kibana. See #113600.

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/210_logsdb_default_mapping.yml [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      catch: bad_request
+      indices.put_index_template:
+        name: "logsdb-index-template"
+        body:
+          index_patterns: ["logsdb"]
+          data_stream: {}
+          composed_of: ["logsdb-mappings"]
+
+  - match: { error.type: "illegal_argument_exception" }
+  - match: { error.reason: "composable template [logsdb-index-template] template after composition with component templates [logsdb-mappings] is invalid" }
```
Comment (kkrik-es) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/210_logsdb_default_mapping.yml:254]: This is not that informative.. any chance we can catch this and complain about @timestamp?

Comment (salvatore-campagna) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/210_logsdb_default_mapping.yml:254]: The is just the top level exception message. There is another one saying
```
The timestamp field [@timestamp] does not exist in the data stream
```

I rpefer not to change this as part of this PR...this tests just capture whatever the status is now.

Comment (martijnvg) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/210_logsdb_default_mapping.yml:254]: This PR is about adding tests for what current expected behavior is. We can consider improving the error message in a follow up PR. I suspect this error is data stream related and not logsdb related. There is data stream logic in place around `@timestamp`.

**Review by martijnvg — APPROVED** [review]
I think we should move this new yaml file to logsdb project. I don't think these very specific tests need to be part of yaml spec rest tests.

LGTM otherwise.

**Comment by salvatore-campagna** [comment]
> I think we should move this new yaml file to logsdb project. I don't think these very specific tests need to be part of yaml spec rest tests.
> 
> LGTM otherwise.

Ok I will move these tests there.

**Comment by salvatore-campagna** [comment]
@martijnvg I moved yaml rest tests to the losgdb xpack plugin.

File: x-pack/plugin/logsdb/src/yamlRestTest/java/org/elasticsearch/xpack/logsdb/LogsDBClientYamlTestSuiteIT.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.LocalClusterSpecBuilder;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
+import org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase;
+import org.junit.ClassRule;
+
+import static org.elasticsearch.test.cluster.FeatureFlag.FAILURE_STORE_ENABLED;
+
+public class LogsDBClientYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
```
Comment (martijnvg) [inline x-pack/plugin/logsdb/src/yamlRestTest/java/org/elasticsearch/xpack/logsdb/LogsDBClientYamlTestSuiteIT.java:23]: Why not reuse `LogsdbTestSuiteIT` class?

Comment (salvatore-campagna) [reply x-pack/plugin/logsdb/src/yamlRestTest/java/org/elasticsearch/xpack/logsdb/LogsDBClientYamlTestSuiteIT.java:23]: I didn't see it was there...I had to merge `main`

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Comment by martijnvg** [comment]
I think this should also be back ported to 8.x?

**Comment by salvatore-campagna** [comment]
> I think this should also be back ported to 8.x?

Yes

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Comment by salvatore-campagna** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/114455)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**